resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -455,7 +456,8 @@ private[spark] class ApplicationMaster(
       val executorMemory = _sparkConf.get(EXECUTOR_MEMORY).toInt
       val executorCores = _sparkConf.get(EXECUTOR_CORES)
       val dummyRunner = new ExecutorRunnable(None, yarnConf, _sparkConf, driverUrl, "<executorId>",
-        "<hostname>", executorMemory, executorCores, appId, securityMgr, localResources)
+        "<hostname>", executorMemory, executorCores, appId, securityMgr, localResources,
+        ResourceProfile.DEFAULT_RESOURCE_PROFILE_ID)
```

Comment:
Just curious if we are envisioning cases where master might also need resources - for example, for the non-parallelizable code ? Is this already handled in your full diff Tom (I did not go over it in detail, sorry), or something we can be a future enhancement if there is need ?

Reply:
so I do not cover allowing the master to change its resource profile as currently that would require it to be killed and restarted.
The driver can request extra resources like GPUs, fpgas, etc via the configs spark.driver.resource.*.    

Can you expand on the non-parallelizable code use case?
 When you say master, do you mean the Driver/ApplicationMaster (same process in cluster mode), or in client mode application master can be separate process, or both?

Reply:
I meant driver. Yes, this will need to be specified a-priori at launch time and cant be dynamically changed. Makes more sense in cluster mode (in client mode, it is the local node of user/notebook server typically).

By non-parallelizable code, I meant what driver runs as part of computation which is not executed in executors - for example, error computation in ML loop, etc.

spark.driver.resource.*, applied  to AM in cluster mode, is exactly what I was looking for -  thanks.